p2p/host/basic/basic_host.go

```diff
@@ -304,11 +309,11 @@ func (h *BasicHost) updateLocalIpAddr() {
 	// them.
 	if len(h.filteredInterfaceAddrs) == 0 {
 		// Add all addresses.
-		h.filteredInterfaceAddrs = append(h.filteredInterfaceAddrs, ifaceAddrs...)
+		h.filteredInterfaceAddrs = h.allInterfaceAddrs
```

Comment:
Are you sure you want an assignment instead of a copy here? It may not hurt to do an assignment here since the function is over, but figured I'd check since a future modification could cause problems.

Reply:
This should be fine.